lib/api/apiUtils/object/objectRestore.js

```diff
@@ -0,0 +1,141 @@
+                    if (err) {
+                        log.trace('request authorization failed', { method: METHOD, error: err });
+                        return next(err);
+                    }
+                    // Call back error if object metadata could not be obtained
+                    if (!objectMD) {
+                        const err = decodedVidResult ? errors.NoSuchVersion : errors.NoSuchKey;
+                        log.trace('error no object metadata found', { method: METHOD, error: err });
+                        return next(err, bucketMD);
+                    }
+                    // If object metadata is delete marker,
+                    // call back NoSuchKey or MethodNotAllowed depending on specifying versionId
+                    if (objectMD.isDeleteMarker) {
+                        let err = errors.NoSuchKey;
+                        if (decodedVidResult) {
+                            err = errors.MethodNotAllowed;
+                        }
+                        log.trace('version is a delete marker', { method: METHOD, error: err });
+                        return next(err, bucketMD, objectMD);
+                    }
+                    log.info('it acquired the object metadata.', {
+                        'method': METHOD,
+                    });
+                    return next(null, bucketMD, objectMD);
+                });
+            },
+
+            // generate restore param obj from xml of request body
+            function parseRequestXml(bucketMD, objectMD, next) {
+                log.trace('parsing object restore information');
+                return parseRestoreRequestXml(request.post, log, (err, restoreInfo) => {
+                    if (err) {
+                        return next(err, bucketMD, objectMD, restoreInfo);
+                    }
+                    log.info('it parsed xml of the request body.', { method: METHOD, value: restoreInfo });
+                    return next(null, bucketMD, objectMD, restoreInfo);
+                });
+            },
+            function checkTierSupported(bucketMD, objectMD, restoreInfo, next) {
+                if (!allowedRestoreObjectRequestTierValues.includes(restoreInfo.tier)) {
+                    return next(errors.NotImplemented, bucketMD, objectMD);
+                }
+                return next(null, bucketMD, objectMD, restoreInfo);
+            },
+            // start restore process
+            function startRestore(bucketMD, objectMD, restoreInfo, next) {
+                return coldStorage.startRestore(bucketMD, objectMD, restoreInfo, log,
+                    (err, _isObjectRestored) => {
+                        isObjectRestored = _isObjectRestored;
+                        return next(err, bucketMD, objectMD);
+                    });
+            },
+            function updateObjectMD(bucketMD, objectMD, next) {
+                metadata.putObjectMD(bucketMD.getName(), objectKey, objectMD, { archive: objectMD.archive },
```

Comment:
The fourth argument of the `putObjectMD`  is the "versioning parameters". The archive's information has already been updated in `startRestore`.

Reply:
done: https://github.com/scality/cloudserver/pull/4507/commits/4bf88225092246150b1f22c1fba17a36d6b0a5f1

Reply:
done: https://github.com/scality/cloudserver/pull/4507/commits/8a706cef41d9750b815dbb593a49b301bd271a26